circuits/cpp/src/aztec3/circuits/kernel/private/native_private_kernel_circuit.cpp

```diff
@@ -301,6 +303,42 @@ void validate_inputs(DummyComposer& composer, PrivateInputs<NT> const& private_i
     NT::fr start_public_call_stack_length = array_length(start.public_call_stack);
     NT::fr start_l1_msg_stack_length = array_length(start.l1_msg_stack);
 
+    {
+        // Verify public key hash matches ethereum address of the sender
+        // TODO(Suyash): Do we want to perform this check only for base case?
```

Comment:
Correct - it's only needed for the base case. (For the "initial private kernel circuit", when we create it).
A signed tx request won't even be sent to the 'non initial' version of the private kernel circuit.

Reply:
right, thanks. I've added a comment to move this to "initial" pkc once its ready.